src/main/java/bio/terra/flight/file/ingest/IngestFileObjectIdStep.java

```diff
@@ -0,0 +1,33 @@
+package bio.terra.flight.file.ingest;
+
+import bio.terra.flight.file.FileMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+
+import java.util.UUID;
+
+// The sole purpose of this step is to allocate the file object id
+// and store it in the working map. That allows the later steps to
+// be simpler. Implementations of file storage on other platforms may
+// be quite different, so we do not want to push this out of the firestore-specific
+// code and into the common code.
+public class IngestFileObjectIdStep implements Step {
+
+    public IngestFileObjectIdStep() {
```

Comment:
Yes. I need to get the object Id into the working map before I do anything else.
Otherwise, the undo doesn't know the object id to clean up. If I store the object Id into the working map during the step, it doesn't get persisted, so on a failure, the undo step wouldn't find it. QED